### PR TITLE
New mqtt implementation

### PR DESCRIPTION
**Pull request by Benzer1406**
Additional output signature changes, documentation corrected and UX fixes.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "tabWidth": 2,
   "useTabs": false,
   "semi": true,
-  "singleQuote": true,
+  "singleQuote": false,
```
Comment (d4vsanchez) [inline .prettierrc:6]: We're currently using single quotes in our project. Was this change made for something I may not be aware of? You can see that multiple changes in this PR is your code changing from single quotes to double quotes.

File: ubidots/ubidots_in.html [hunk 2]
```diff
@@ -50,78 +50,78 @@
       </div>
       <!-- VARIABLE 1 -->
       <div id="variable-1" class="form-row" style="margin-bottom:10px">
```
Comment (d4vsanchez) [inline ubidots/ubidots_in.html:52]: Why not changing this to a flexbox container so we can center everything vertically without any hassle?

File: ubidots/ubidots_in.html [hunk 3]
```diff
@@ -50,78 +50,78 @@
...
-        <label style="width:24px" id=node-input-variable-1-space></label>
-         <input type="text" style="width: calc(100% - 250px); margin-left:28px" id="node-input-label_variable_1" />
+         <label style="width:23px" id=node-input-variable-1-space></label>
+         <input type="text" style="width: calc(100% - 250px)" id="node-input-label_variable_1" />
```
Comment (d4vsanchez) [inline ubidots/ubidots_in.html:54]: With the flexbox container you could only do `flex: 1` here and forget about calculating the width that way.

File: ubidots/ubidots_in.html [hunk 4]
```diff
@@ -50,78 +50,78 @@
       </div>
       <!-- VARIABLE 1 -->
       <div id="variable-1" class="form-row" style="margin-bottom:10px">
-        <label style="width:24px" id=node-input-variable-1-space></label>
-         <input type="text" style="width: calc(100% - 250px); margin-left:28px" id="node-input-label_variable_1" />
+         <label style="width:23px" id=node-input-variable-1-space></label>
```
Comment (d4vsanchez) [inline ubidots/ubidots_in.html:53]: With flex you can remove the width from here.

File: ubidots/ubidots_in.html [hunk 5]
```diff
@@ -50,78 +50,78 @@
...
+         <input type="text" style="width: calc(100% - 250px)" id="node-input-label_variable_1" />
          <input class="variable__lastValueCheckbox" type="checkbox" checked id="node-input-checkbox_variable_1_last_value" >
-          <label id="node-input-last-value-label-1" style="width: 89px" data-i18n="ubidots.lastValue"></label>
+         <label id="node-input-last-value-label-1" style="width: 89px" data-i18n="ubidots.lastValue"></label>
```
Comment (d4vsanchez) [inline ubidots/ubidots_in.html:56]: And also you could remove the width from here.

File: ubidots/ubidots_in.js [hunk 6]
```diff
@@ -90,6 +89,31 @@ module.exports = function (RED) {
         }
       });
     });
+
+    function defineOutputObject(topic, message) {
+      let finalObject = {};
+      let variable = topic;
+      if (useCustomTopics) {
+        variable = topic.substring(14);
+        if (topic.substring(topic.length - 3) === "/lv") {
```
Comment (d4vsanchez) [inline ubidots/ubidots_in.js:98]: You can simplify this by using [String.prototype.endsWith](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith)

File: ubidots/ubidots_in.js [hunk 7]
```diff
@@ -90,6 +89,31 @@ module.exports = function (RED) {
...
+          finalObject[variable] = { value: JSON.parse(message.toString()) };
+        } else {
+          finalObject[variable] = JSON.parse(message.toString());
+        }
+      } else {
+        if (topic.substring(topic.length - 3) === "/lv") {
```
Comment (d4vsanchez) [inline ubidots/ubidots_in.js:104]: You can simplify this by using [String.prototype.endsWith](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith)

File: ubidots/ubidots_in.js [hunk 8]
```diff
@@ -90,6 +89,31 @@ module.exports = function (RED) {
...
+          variable = topic.slice(0, topic.length - 3);
+        }
+        let array = variable.split("/");
+        variable = array[array.length - 1];
+        if (topic.substring(topic.length - 3) === "/lv") {
```
Comment (d4vsanchez) [inline ubidots/ubidots_in.js:109]: You can simplify this by using [String.prototype.endsWith](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith)

File: ubidots/ubidots_in.js [hunk 9]
```diff
@@ -90,6 +89,31 @@ module.exports = function (RED) {
...
+          finalObject[variable] = { value: JSON.parse(message.toString()) };
+        } else {
+          finalObject[variable] = JSON.parse(message.toString());
+        }
+      } else {
+        if (topic.substring(topic.length - 3) === "/lv") {
+          variable = topic.slice(0, topic.length - 3);
+        }
+        let array = variable.split("/");
```
Comment (d4vsanchez) [inline ubidots/ubidots_in.js:107]: What do you think about naming this variable a bit more explicit? I mean, the name could say what's the data that this array has.

File: ubidots/ubidots_in.js [hunk 10]
```diff
@@ -90,6 +89,31 @@ module.exports = function (RED) {
...
+    function defineOutputObject(topic, message) {
+      let finalObject = {};
+      let variable = topic;
+      if (useCustomTopics) {
+        variable = topic.substring(14);
+        if (topic.substring(topic.length - 3) === "/lv") {
+          finalObject[variable] = { value: JSON.parse(message.toString()) };
+        } else {
+          finalObject[variable] = JSON.parse(message.toString());
+        }
```
Comment (d4vsanchez) [inline ubidots/ubidots_in.js:102]: I see that these lines are the same as the ones from +109 to +113. What do you think about creating a function so you don't need to duplicate them?

**Comment by Benzer1406** [comment]
Changes applied and issues fixed @d4vsanchez

**Comment by d4vsanchez** [comment]
@Benzer1406 Approved, :sparkles: nice job :sparkles:

I'm going to merge it.

**Pull request merged**